Seed data

3oheil opened the pull request:
Seed Data in Author Table

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Introduced automatic seeding of initial data from Excel files on application startup, including authors, books, categories, keywords, and their relationships.
  * Added support for reading and importing data from Excel files for streamlined bulk data entry.

* **Database Changes**
  * Added a new Authors table and established a many-to-many relationship between books and authors.
  * Enforced unique slugs for books and authors to ensure data integrity.

* **Chores**
  * Integrated the EPPlus package for Excel file processing.
  * Included sample Excel seed data files in the project output.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

3oheil11 committed:
add AuthorData.xlsx to SeedData Foulder

3oheil11 committed:
Move Seed Data foulder to main root project

3oheil11 committed:
create foulder to write class

3oheil11 committed:
create ExcelSeeder class

3oheil11 committed:
design and write ExcelSeeder class

3oheil11 committed:
create and design Seed Data Configuration in Program File

3oheil11 committed:
Apply changes

3oheil11 committed:
add migration

3oheil11 committed:
add migration

3oheil11 committed:
snapahot updated

3oheil11 committed:
apply changes

3oheil11 committed:
apply changes and install Package EPPlus version 5.8.3

3oheil11 committed:
Add Book Data Excel file

3oheil11 committed:
add comment to ReadAuthorsFromExcel Method

3oheil11 committed:
add comment to ReadBooksFromExcel Method

3oheil11 committed:
create ReadBooksFromExcel Method

3oheil11 committed:
This class is designed to separate configurations in the program.

3oheil11 committed:
Rename

3oheil11 committed:
Rename class

3oheil11 committed:
add Authors and Books Configuration

3oheil11 committed:
add SeedInitialData to Program

3oheil11 committed:
add AuthorBook Data in excel file

3oheil11 committed:
seed log

3oheil11 committed:
add Book Keyword wxcel file

3oheil11 committed:
add book relation excel file

3oheil11 committed:
add category excel file

3oheil11 committed:
add keyword excel file

3oheil11 committed:
add ReadKeywordFromExcel method

3oheil11 committed:
add ReadCategoryFromExcel Method

3oheil11 committed:
add ReadBookRelationFromExcel method

3oheil11 committed:
add ReadBookKeywordFromExcel Method

3oheil11 committed:
Apply changes

3oheil11 committed:
Updata solution

3oheil11 committed:
update class

3oheil11 committed:
add comment for authors

3oheil11 committed:
add comment for Book

3oheil11 committed:
add AuthorBook Initial

3oheil11 committed:
add BookKeyword initial

3oheil11 committed:
book relational initial

3oheil11 committed:
add All Initialize

hootanht committed:
Update ci.yml (#100) (#104)

Mohammad-Faraji committed:
Fix Bug File DataSeeder And ExcelSeeder

Mohammad-Faraji committed:
Conflict resolution

Mohammad-Faraji committed:
Update Namespace

hootanht commented on Refhub/seed-log.txt:
```diff
@@ -0,0 +1,5 @@
+[6/17/2025 2:52:18 AM] Starting seeding...
```
I think we don't want that log file anymore here.
@3oheil

hootanht commented:
@coderabbitai full review

hootanht commented:
/gemini review

hootanht committed:
Update project documentation and refactor codebase

Enhanced `copilot-instructions.md` with project overview, architecture, and workflows. Updated `.gitignore` to exclude `MigrationBackup/`. Modified `mcp.json` for new server configurations.

Refactored views (`Index.cshtml`, `Update.cshtml`, `BestBookView.cshtml`, `LastBookView.cshtml`) for improved layout and image handling. Added validation scripts in `_ValidationScriptsPartial.cshtml`.

Implemented new API endpoints in `AuthorsController.cs` and `BooksController.cs`, and improved file handling in `BookController.cs`. Introduced `S3FileUploaderService.cs` for S3 file management.

Expanded `README.md` with detailed project documentation and created additional markdown files for specific features. Improved project structure and added new enums and configuration classes.

Updated `appsettings.Development.json` and `launchSettings.json` for better environment management. Enhanced `Refhub.csproj` with new package references and improved localization in `Messages.Designer.cs`.

Overall, the codebase has been refactored for better readability, maintainability, and performance, adhering to ASP.NET Core best practices.

hootanht committed:
Enhance data seeding with error handling and logging

- Added error handling in `DataSeeder.cs` to log exceptions and prevent duplicate entries during seeding.
- Improved data reading methods in `ExcelSeeder.cs` with file existence checks and better error handling.
- Created new `Authors` and `BookAuthors` tables in the migration file `20250616145440_AddAuthorTable.cs`.
- Updated `Program.cs` to include error handling for initial data seeding.
- Added `EPPlus` package reference in `Refhub.csproj` for Excel file operations.
- Revised `seed-log.txt` to improve logging format and detail.
